akkaClient/src/main/scala/ca/pigscanfly/SwarmMessageClient.scala

```diff
@@ -0,0 +1,67 @@
+package ca.pigscanfly
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import ca.pigscanfly.httpClient.HttpClient
+import ca.pigscanfly.models.MessagePost.encoder
+import ca.pigscanfly.models.MessageRetrieval._
+import ca.pigscanfly.models.{Message, MessageDelivery, MessagePost, MessageRetrieval}
+import io.circe.parser
+import io.circe.syntax._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait SwarmMessageClient {
+  this: HttpClient =>
+  implicit def actorSystem: ActorSystem
+
+  implicit def executionContext: ExecutionContext
+
+  def getMessages(url: String, headers: List[HttpHeader]): Future[MessageRetrieval] = {
+    for {
+      response <- sendRequest(HttpRequest(uri = url, headers = headers))
+      messagesString <- Unmarshal(response.entity).to[String]
+      messages <- parser.decode[List[Message]](messagesString) match {
+        case Right(messages) =>
+          Future(messages.map(message => message.copy(data = new String(java.util.Base64.getDecoder.decode(message.data)))))
```

Comment:
This is the part we should decode with protocolbuffer

Reply:
Noted